apps/astarte_data_updater_plant/test/astarte_data_updater_plant/data_updater_test.exs

```diff
@@ -1882,6 +1913,27 @@ defmodule Astarte.DataUpdaterPlant.DataUpdaterTest do
     ensure_message_has_been_handled(realm, device_id)
   end
 
+  defp handle_capabilities(realm, encoded_device_id, payload, _tracking_id, timestamp) do
+    {:ok, device_id} = Astarte.Core.Device.decode_device_id(encoded_device_id)
+
+    headers =
+      headers_fixture(realm, encoded_device_id, x_astarte_msg_type: "capabilities")
+
+    publish_opts = [
+      headers: headers,
+      message_id: generate_message_id(realm, encoded_device_id, timestamp),
+      timestamp: timestamp,
+      sharding_key: {realm, device_id}
+    ]
+
+    {:ok, pid} = Mississippi.Consumer.MessageTracker.get_message_tracker({realm, device_id})
+    :erlang.trace(pid, true, [:receive])
+
+    :ok = Mississippi.Producer.EventsProducer.publish(payload, publish_opts)
+
+    assert_receive {:trace, ^pid, :receive, {_, {_, _}, {:ack_delivery, _message}}}
```

Comment:
Is there a reason to write this instead of using the [`ensure_message_has_been_handled/2`](https://github.com/astarte-platform/astarte/blob/7e10975abf83518f4ae7de52e1a6150bdf404f9a/apps/astarte_data_updater_plant/test/astarte_data_updater_plant/data_updater_test.exs#L1965-L1970) helper?

Reply:
In theory sending the message after starting the message tracker should prevent some race conditions from happening (the message tracker is too slow to get up and the message gets handled before it can be recorded)